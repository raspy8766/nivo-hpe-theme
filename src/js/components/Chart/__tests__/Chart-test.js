import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Grommet } from '../../Grommet';
import { Box } from '../../Box';
import { Chart, calcs } from '..';

const VALUES = [
  { value: [1, 60], label: 'sixty' },
  { value: [0, 0], label: 'zero' },
];

const UNDEFINED_VALUES = [
  { value: [2, 60], label: 'sixty' },
  { value: [1, undefined] },
  { value: [0, 0], label: 'zero' },
];

describe('Chart', () => {
  test('default', () => {
    const component = renderer.create(
      <Grommet>
        <Chart values={VALUES} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('type', () => {
    const component = renderer.create(
      <Grommet>
        <Chart type="bar" values={VALUES} />
        <Chart type="line" values={VALUES} />
        <Chart type="area" values={VALUES} />
        <Chart type="point" values={VALUES} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('size', () => {
    const component = renderer.create(
      <Grommet>
        <Chart size="xsmall" values={VALUES} />
        <Chart size="small" values={VALUES} />
        <Chart size="medium" values={VALUES} />
        <Chart size="large" values={VALUES} />
        <Chart size="xlarge" values={VALUES} />
        <Box width="large">
          <Chart size={{ width: 'full' }} values={VALUES} />
          <Chart size={{ width: 'fill' }} values={VALUES} />
          <Chart size={{ width: 'auto' }} values={VALUES} />
        </Box>
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('thickness', () => {
    const component = renderer.create(
      <Grommet>
        <Chart thickness="xsmall" values={VALUES} />
        <Chart thickness="small" values={VALUES} />
        <Chart thickness="medium" values={VALUES} />
        <Chart thickness="large" values={VALUES} />
        <Chart thickness="xlarge" values={VALUES} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('cap', () => {
    const component = renderer.create(
      <Grommet>
        <Chart round values={VALUES} />
        <Chart type="line" round values={VALUES} />
        <Chart type="area" round values={VALUES} />
        <Chart type="point" round values={VALUES} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('gap', () => {
    const component = renderer.create(
      <Grommet>
        <Box width="large">
          <Chart size={{ width: 'auto' }} gap="small" values={VALUES} />
          <Chart size={{ width: 'auto' }} gap="medium" values={VALUES} />
          <Chart size={{ width: 'auto' }} gap="large" values={VALUES} />
        </Box>
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('dash', () => {
    const component = renderer.create(
      <Grommet>
        <Chart dash values={VALUES} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('color', () => {
    const component = renderer.create(
      <Grommet>
        <Chart color="brand" values={VALUES} />
        <Chart color={{ color: 'brand', opacity: 'strong' }} values={VALUES} />
        <Chart
          color={[
            { value: 0, color: 'brand' },
            { value: 60, color: 'border' },
          ]}
          values={VALUES}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('pad', () => {
    const component = renderer.create(
      <Grommet>
        <Chart pad="xsmall" values={VALUES} />
        <Chart
          pad={{ horizontal: 'medium', vertical: 'small' }}
          values={VALUES}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('undefined values', () => {
    const component = renderer.create(
      <Grommet>
        <Chart type="bar" values={UNDEFINED_VALUES} />
        <Chart type="line" values={UNDEFINED_VALUES} />
        <Chart type="area" values={UNDEFINED_VALUES} />
        <Chart type="point" values={UNDEFINED_VALUES} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calcs', () => {
    const result = calcs([1, 2, 3]);
    expect(result).toMatchSnapshot();
  });
});
