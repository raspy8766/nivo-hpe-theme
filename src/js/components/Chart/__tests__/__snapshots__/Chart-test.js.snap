// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart calcs 1`] = `
Object {
  "axis": Array [
    Array [
      0,
      2,
    ],
    Array [
      3.2,
    ],
  ],
  "bounds": Array [
    Array [
      0,
      2,
    ],
    Array [
      0.8,
      3.2,
    ],
  ],
  "dimensions": Array [
    2,
    2.4000000000000004,
  ],
  "pad": "large",
  "thickness": "xlarge",
}
`;

exports[`Chart cap 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <path
          d="M 384,0 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="#6FFFB0"
      stroke="#6FFFB0"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={24}
    >
      <g>
        <path
          d="M 384,0 L 0,192 L 0,192 L 384,192 Z"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="#6FFFB0"
      stroke="none"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <g
        stroke="none"
      >
        <title>
          sixty
        </title>
        <circle
          cx={384}
          cy={0}
          r={12}
        />
      </g>
      <g
        stroke="none"
      >
        <title>
          zero
        </title>
        <circle
          cx={0}
          cy={192}
          r={12}
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Chart color 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#7D4CDB"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      opacity={0.8}
      stroke="#7D4CDB"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <defs>
      <linearGradient
        id="undefined-gradient"
        x1={0}
        x2={0}
        y1={0}
        y2={1}
      >
        <stop
          offset={0}
          stopColor="rgba(0,0,0,0.33)"
        />
        <stop
          offset={1}
          stopColor="#7D4CDB"
        />
      </linearGradient>
      <mask
        id="undefined-mask"
      >
        <g
          fill="none"
          stroke="#ffffff"
          strokeLinecap="butt"
          strokeLinejoin="miter"
          strokeWidth={24}
        >
          <g
            fill="none"
          >
            <title>
              sixty
            </title>
            <path
              d="M 384,192 L 384,0"
            />
          </g>
          <g
            fill="none"
          >
            <title>
              zero
            </title>
            <path
              d="M 0,192 L 0,192"
            />
          </g>
        </g>
      </mask>
    </defs>
    <rect
      fill="url(#undefined-gradient)"
      height={216}
      mask="url(#undefined-mask)"
      width={408}
      x={-12}
      y={-12}
    />
  </svg>
</div>
`;

exports[`Chart dash 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
          strokeDasharray="48 12"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
          strokeDasharray="48 12"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Chart default 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Chart gap 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 768px;
}

.c2 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      height={192}
      preserveAspectRatio="none"
      viewBox="-12 -12 84 216"
      width={60}
    >
      <g
        fill="none"
        stroke="#6FFFB0"
        strokeLinecap="butt"
        strokeLinejoin="miter"
        strokeWidth={24}
      >
        <g
          fill="none"
        >
          <title>
            sixty
          </title>
          <path
            d="M 60,192 L 60,0"
          />
        </g>
        <g
          fill="none"
        >
          <title>
            zero
          </title>
          <path
            d="M 0,192 L 0,192"
          />
        </g>
      </g>
    </svg>
    <svg
      className="c2"
      height={192}
      preserveAspectRatio="none"
      viewBox="-12 -12 96 216"
      width={72}
    >
      <g
        fill="none"
        stroke="#6FFFB0"
        strokeLinecap="butt"
        strokeLinejoin="miter"
        strokeWidth={24}
      >
        <g
          fill="none"
        >
          <title>
            sixty
          </title>
          <path
            d="M 72,192 L 72,0"
          />
        </g>
        <g
          fill="none"
        >
          <title>
            zero
          </title>
          <path
            d="M 0,192 L 0,192"
          />
        </g>
      </g>
    </svg>
    <svg
      className="c2"
      height={192}
      preserveAspectRatio="none"
      viewBox="-12 -12 120 216"
      width={96}
    >
      <g
        fill="none"
        stroke="#6FFFB0"
        strokeLinecap="butt"
        strokeLinejoin="miter"
        strokeWidth={24}
      >
        <g
          fill="none"
        >
          <title>
            sixty
          </title>
          <path
            d="M 96,192 L 96,0"
          />
        </g>
        <g
          fill="none"
        >
          <title>
            zero
          </title>
          <path
            d="M 0,192 L 0,192"
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`Chart pad 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-18 -18 420 228"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-36 -24 456 240"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Chart size 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 768px;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={96}
    preserveAspectRatio="none"
    viewBox="-12 -12 120 120"
    width={96}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 96,96 L 96,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,96 L 0,96"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 216 216"
    width={192}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 192,192 L 192,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={384}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 408"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,384 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,384 L 0,384"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={768}
    preserveAspectRatio="none"
    viewBox="-12 -12 792 792"
    width={768}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 768,768 L 768,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,768 L 0,768"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={1152}
    preserveAspectRatio="none"
    viewBox="-12 -12 1176 1176"
    width={1152}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 1152,1152 L 1152,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,1152 L 0,1152"
        />
      </g>
    </g>
  </svg>
  <div
    className="c2"
  >
    <svg
      className="c1"
      height={192}
      preserveAspectRatio="none"
      viewBox="-12 -12 24 216"
      width={0}
    >
      <g
        fill="none"
        stroke="#6FFFB0"
        strokeLinecap="butt"
        strokeLinejoin="miter"
        strokeWidth={24}
      >
        <g
          fill="none"
        >
          <title>
            sixty
          </title>
          <path
            d="M 0,192 L 0,0"
          />
        </g>
        <g
          fill="none"
        >
          <title>
            zero
          </title>
          <path
            d="M 0,192 L 0,192"
          />
        </g>
      </g>
    </svg>
    <svg
      className="c1"
      height={192}
      preserveAspectRatio="none"
      viewBox="-12 -12 24 216"
      width={0}
    >
      <g
        fill="none"
        stroke="#6FFFB0"
        strokeLinecap="butt"
        strokeLinejoin="miter"
        strokeWidth={24}
      >
        <g
          fill="none"
        >
          <title>
            sixty
          </title>
          <path
            d="M 0,192 L 0,0"
          />
        </g>
        <g
          fill="none"
        >
          <title>
            zero
          </title>
          <path
            d="M 0,192 L 0,192"
          />
        </g>
      </g>
    </svg>
    <svg
      className="c1"
      height={192}
      preserveAspectRatio="none"
      viewBox="-12 -12 96 216"
      width={72}
    >
      <g
        fill="none"
        stroke="#6FFFB0"
        strokeLinecap="butt"
        strokeLinejoin="miter"
        strokeWidth={24}
      >
        <g
          fill="none"
        >
          <title>
            sixty
          </title>
          <path
            d="M 72,192 L 72,0"
          />
        </g>
        <g
          fill="none"
        >
          <title>
            zero
          </title>
          <path
            d="M 0,192 L 0,192"
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`Chart thickness 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-3 -3 390 198"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={6}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-6 -6 396 204"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={12}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-24 -24 432 240"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={48}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-48 -48 480 288"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={96}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Chart type 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <path
          d="M 384,0 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="#6FFFB0"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g>
        <path
          d="M 384,0 L 0,192 L 0,192 L 384,192 Z"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="#6FFFB0"
      stroke="none"
      strokeLinecap="butt"
      strokeLinejoin="miter"
    >
      <g
        stroke="none"
      >
        <title>
          sixty
        </title>
        <rect
          height={24}
          width={24}
          x={372}
          y={-12}
        />
      </g>
      <g
        stroke="none"
      >
        <title>
          zero
        </title>
        <rect
          height={24}
          width={24}
          x={-12}
          y={180}
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Chart undefined values 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: block;
  max-width: 100%;
  overflow: visible;
}

<div
  className="c0"
>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <title>
          sixty
        </title>
        <path
          d="M 384,192 L 384,0"
        />
      </g>
      <g
        fill="none"
      >
        <title>
          zero
        </title>
        <path
          d="M 0,192 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="none"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g
        fill="none"
      >
        <path
          d="M 384,0 L 0,192"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="#6FFFB0"
      stroke="#6FFFB0"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeWidth={24}
    >
      <g>
        <path
          d="M 384,0 L 0,192 L 0,192 L 384,192 Z"
        />
      </g>
    </g>
  </svg>
  <svg
    className="c1"
    height={192}
    preserveAspectRatio="none"
    viewBox="-12 -12 408 216"
    width={384}
  >
    <g
      fill="#6FFFB0"
      stroke="none"
      strokeLinecap="butt"
      strokeLinejoin="miter"
    >
      <g
        stroke="none"
      >
        <title>
          sixty
        </title>
        <rect
          height={24}
          width={24}
          x={372}
          y={-12}
        />
      </g>
      <g
        stroke="none"
      >
        <title>
          zero
        </title>
        <rect
          height={24}
          width={24}
          x={-12}
          y={180}
        />
      </g>
    </g>
  </svg>
</div>
`;
